instrumentation/aws_sdk/Appraisals

```diff
@@ -4,58 +4,27 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-appraise 'aws-sdk-3.1' do
-  gem 'aws-sdk', '~> 3.1'
+appraise 'aws-sdk-3' do
+  gem 'aws-sdk-core', '~> 3'
+  gem 'aws-sdk-lambda', '~> 1'
+  gem 'aws-sdk-dynamodb', '~> 1'
+  gem 'aws-sdk-sns', '~> 1'
+  gem 'aws-sdk-sqs', '~> 1'
 end
```

Comment:
The recent version of `aws-sdk-3.x` does not contain the recent gem releases so I omitted these sets and just included the relevant gems for testing.

Also reduced the number of V2 appraisal definitions to just 2.

Reply:
Additional benefit - the aws-sdk gem (which we dont really support anymore) also brings in tons of additional dependencies (almost 400 service gems at this point).